internal/mode/static/nginx/runtime/runtime_manager_test.go

```diff
@@ -0,0 +1,95 @@
+package runtime_test
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/nginx/runtime"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/nginx/runtime/runtimefakes"
+	ngxclient "github.com/nginxinc/nginx-plus-go-client/client"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("NGINX Runtime Manager", Ordered, func() {
```

Comment:
Please put these tests in the `manager_test.go` file, under the existing Describe block.

Reply:
I have tried already but had issue with import cycle since we are testing real Manager and when I put this test into manager_test.go, there's import cycle since tests are under runtime, so I added them into another file with pckg runtime_test under "runtime_test", I see your comment on this before, but Im not sure am I missing on something, do you want me to update manager_test.go pckg to "runtime_test", even tho that creates an issue within "findMainProcess" fnction that's not being exported, and by doing so will also have to update that with both tests to "FindMainProcess", what do you think ? maybe Im missing onto something

Reply:
Which import causes the cycle? Is it because you are importing `runtime`? You shouldn't need to import that if we are already in that package. We can just call `NewManagerImpl` directly.

Reply:
sounds good, and definitely that one could be updated as you advise, will do !
 
but actually this one was an issue: 
"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/nginx/runtime/runtimefakes"

manager_test.go is under `runtime`, I can update manager_test.go to be under `runtime_test` ?

Reply:
sounds good, I will add corresponding update

Reply:
That `/proc` dir exists in the nginx container in a real environment, regardless of which OS it's run on. The reason it's failing here is because we're running unit tests on our local machine, where the file obviously doesn't exist.

I think having a wrapper for os.ReadFile is probably fine in this case.

Reply:
thanks for explanation, that makes sense, but if you think we can do something other about ReadFile I can align